aws/data_source_aws_region_test.go

```diff
@@ -89,197 +83,110 @@ func TestAccDataSourceAwsRegion_basic(t *testing.T) {
 			{
 				Config: testAccDataSourceAwsRegionConfig_empty,
 				Check: resource.ComposeTestCheckFunc(
-					testAccDataSourceAwsRegionCheck(resourceName),
-					resource.TestCheckResourceAttr(resourceName, "endpoint", "ec2.us-east-1.amazonaws.com"),
-					resource.TestCheckResourceAttr(resourceName, "name", "us-east-1"),
-					resource.TestCheckResourceAttr(resourceName, "description", "US East (N. Virginia)"),
+					resource.TestMatchResourceAttr(dataSourceName, "description", regexp.MustCompile(`^.+$`)),
+					testAccCheckResourceAttrRegionalHostnameService(dataSourceName, "endpoint", ec2.EndpointsID),
+					resource.TestCheckResourceAttr(dataSourceName, "name", testAccGetRegion()),
 				),
 			},
 		},
 	})
 }
 
 func TestAccDataSourceAwsRegion_endpoint(t *testing.T) {
-	// Ensure we always get a consistent result
-	oldvar := os.Getenv("AWS_DEFAULT_REGION")
-	os.Setenv("AWS_DEFAULT_REGION", "us-east-1")
-	defer os.Setenv("AWS_DEFAULT_REGION", oldvar)
-
-	endpoint1 := "ec2.us-east-1.amazonaws.com"
-	endpoint2 := "ec2.us-east-2.amazonaws.com"
-	name1 := "us-east-1"
-	name2 := "us-east-2"
-	description1 := "US East (N. Virginia)"
-	description2 := "US East (Ohio)"
-	resourceName := "data.aws_region.test"
+	dataSourceName := "data.aws_region.test"
 
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:  func() { testAccPreCheck(t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccDataSourceAwsRegionConfig_endpoint(endpoint1),
-				Check: resource.ComposeTestCheckFunc(
-					testAccDataSourceAwsRegionCheck(resourceName),
-					resource.TestCheckResourceAttr(resourceName, "endpoint", endpoint1),
-					resource.TestCheckResourceAttr(resourceName, "name", name1),
-					resource.TestCheckResourceAttr(resourceName, "description", description1),
-				),
-			},
-			{
-				Config: testAccDataSourceAwsRegionConfig_endpoint(endpoint2),
+				Config: testAccDataSourceAwsRegionConfig_endpoint(),
 				Check: resource.ComposeTestCheckFunc(
-					testAccDataSourceAwsRegionCheck(resourceName),
-					resource.TestCheckResourceAttr(resourceName, "endpoint", endpoint2),
-					resource.TestCheckResourceAttr(resourceName, "name", name2),
-					resource.TestCheckResourceAttr(resourceName, "description", description2),
+					resource.TestMatchResourceAttr(dataSourceName, "description", regexp.MustCompile(`^.+$`)),
+					resource.TestMatchResourceAttr(dataSourceName, "endpoint", regexp.MustCompile(fmt.Sprintf("^ec2\\.[^.]+\\.%s$", testAccGetPartitionDNSSuffix()))),
```

Comment:
Why not?
```suggestion
					rtestAccCheckResourceAttrRegionalHostnameService(dataSourceName, "endpoint", ec2.EndpointsID),
```

Reply:
The configuration uses the following to allow it to be region-agnostic:

```
endpoint = "ec2.${tolist(data.aws_regions.test.names)[0]}.${data.aws_partition.test.dns_suffix}"
```

The first region is not guaranteed to match the region of the test provider instance due to how the DescribeRegions API works and we probably shouldn't be using `aws_region` in the testing for `aws_region` to try and align things. 😄